examples/mixtral/train_mixtral_moe.sh

```diff
@@ -0,0 +1,400 @@
+echo "AC: $AC"
+echo "RECOMPUTE_METHOD: $RECOMPUTE_METHOD"
+echo "RECOMPUTE_NUM_LAYERS: $RECOMPUTE_NUM_LAYERS"
+echo ""
+
+
+if [ $AC = full ]; then
+    activation_checkpoint_options=(
+		    --recompute-method ${RECOMPUTE_METHOD} 
+		    --recompute-granularity full 
+            --recompute-num-layers ${RECOMPUTE_NUM_LAYERS})
+elif [ $AC = sel ]; then
+    activation_checkpoint_options=(
+        --recompute-activations)
+elif [ $AC = none ]; then
+    activation_checkpoint_options=()
+    
+fi
+
+if [ $USE_GROUPED_GEMM = true ]; then
+    USE_GROUPED_GEMM_OPTION="--moe-grouped-gemm"
+else
+    USE_GROUPED_GEMM_OPTION=""
+fi
+
+if [ $MOE_USE_LEGACY_GROUPED_GEMM = true ]; then
+    USE_LEGACY_GROUPED_GEMM_OPTION="--moe-use-legacy-grouped-gemm"
+else
+    USE_LEGACY_GROUPED_GEMM_OPTION=""
+    # disable gemm tuning when using TE Group GEMM.
+    GEMM_TUNING=0
+    echo "[WARN] GEMM tuning is disabled when using TransformerEngine Group GEMM."
+fi
+
+# gemm tuning, https://github.com/ROCm/TransformerEngine
+if [ "$GEMM_TUNING" -eq 1 ]; then
+   export TE_HIPBLASLT_TUNING_RUN_COUNT=10
+   export TE_HIPBLASLT_TUNING_ALGO_COUNT=50
+else
+   unset TE_HIPBLASLT_TUNING_RUN_COUNT
+   unset TE_HIPBLASLT_TUNING_ALGO_COUNT
+fi
+
+if [ $GPT_LAYER_IN_TE = true ]; then
+    TRANSFORMER_IMPL=transformer_engine
+else
+    TRANSFORMER_IMPL=local
+fi
+
+
+
+if [ -z ${MP_VP} ]; then
+    vp_options=()
+else
+    vp_options=(
+        --num-layers-per-virtual-pipeline-stage ${MP_VP} 
+        )
+fi
+
+if [ "$NVTE_CK_USES_BWD_V3" -eq 1 ]; then
+    echo "Using BWD FAv3"
+    export NVTE_CK_USES_BWD_V3=1    #by default 0, if set to 1, some cases will call the bwd v3 dqdkdv kernel;
+    export NVTE_CK_V3_ATOMIC_FP32=0 #by default 1, if set to 0 will use atomic fp16/bf16(w/o convert_dq kernel) when NVTE_CK_USES_BWD_V3 is set to 1;
+    export NVTE_CK_V3_SPEC=1        #by default 0, if set to 1 will call the specialized v3 kernel when NVTE_CK_USES_BWD_V3 is set to 1;
+else
+    echo "Disabling BWD FAv3"
+    export NVTE_CK_USES_BWD_V3=0
+    export NVTE_CK_V3_ATOMIC_FP32=1
+    export NVTE_CK_V3_SPEC=0
+fi
+
+
+
+DISTRIBUTED_ARGS=(
+    --nproc_per_node $GPUS_PER_NODE
+    --nnodes $NNODES
+    --node_rank $NODE_RANK
+    --master_addr $MASTER_ADDR
+    --master_port $MASTER_PORT
+)
+
+if [ "$MODEL_SIZE" = "8x7B" ]; then
+MODEL_ARGS=(
+    --use-mcore-models
+    --disable-bias-linear
+    --seq-length ${SEQLEN}
+    --max-position-embeddings 32768
+    --num-layers 32
+    --hidden-size 4096
+    --ffn-hidden-size 14336
+    --num-attention-heads 32
+    --init-method-std 0.01
+    --attention-dropout 0.0
+    --hidden-dropout 0.0
+    --normalization RMSNorm
+    --position-embedding-type rope
+    --swiglu
+    --untie-embeddings-and-output-weights
+    --group-query-attention
+    --num-query-groups 8
+    --no-masked-softmax-fusion
+    --no-position-embedding
+    --rotary-base 1000000
+)
+elif [ "$MODEL_SIZE" = "8x22B" ]; then
+MODEL_ARGS=(
```

Comment:
again, there are duplicates in the code. Please refactor to make is more minimal and clean.